File: src/main/java/com/vsu/by/app/education/mistake/Mistake.java
package com.vsu.by.app.education.mistake;

import com.vsu.by.app.education.rule.Rule;
import com.vsu.by.app.education.task.Task;
import lombok.Data;

import javax.persistence.*;

@Data
@Entity
@Table(name = "Mistake")
public class Mistake {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @OneToOne(cascade = {CascadeType.MERGE})
    private Rule rule;
    @OneToOne(cascade = {CascadeType.MERGE})
    private Task task;
    private int wordNo;
    private int symbolNo;
}

File: angular/testing-programm/src/app/mistake/mistake.ts
import { PupilAttempt } from '../pupil-attempt/pupilAttempt';
import { Rule } from '../rule/rule';

export class Mistake {
    id: number;
    pupilAttempt: PupilAttempt;
    rule: Rule;
    wordNo: number;
    symbolNo: number;
}
